[Lalelulilulela] iP

Lalelulilulela opened the pull request:
# DukePro 
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/231).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Lalelulilulela committed:
Rename, Greet, Exit

Lalelulilulela committed:
Echo

Lalelulilulela committed:
Add, List

Lalelulilulela committed:
Mark as Done

Lalelulilulela committed:
ToDos, Events, Deadlines

Lalelulilulela committed:
Added testing

Lalelulilulela committed:
Handle errors

Lalelulilulela committed:
Handle errors

Lalelulilulela committed:
Added delete

Lalelulilulela committed:
Added enums for error types

Lalelulilulela committed:
Add Storage.java

Lalelulilulela committed:
Update Storage.java

Lalelulilulela committed:
Merge branch 'branch-Level-7'

Lalelulilulela committed:
Update Deadline.java to store deadline as Date and Time

Lalelulilulela committed:
Merge branch 'branch-Level-8'

Lalelulilulela committed:
Add UI.java

Lalelulilulela committed:
Add Commands package and Parser.java

Lalelulilulela committed:
Add TaskList.java

Lalelulilulela committed:
Organize classes into packages

Lalelulilulela committed:
Merge commit 'a6f73244a20539e20c2189973ebf597df34ef631'

Lalelulilulela committed:
Fix greet message

Lalelulilulela committed:
Add Gradle

Lalelulilulela committed:
Add JUnit tests

nigel27022001 commented on src/main/java/Exceptions/InvalidInputException.java:
```diff
@@ -0,0 +1,7 @@
+package Exceptions;
+
+public class InvalidInputException extends LeluException{
+    public InvalidInputException(String message) {
+        super(message);
+    }
+}
```
I like how you have created even more specific exceptions ontop of LeluException

nigel27022001 commented on src/main/java/Lelu/Parser.java:
```diff
@@ -0,0 +1,32 @@
+import Commands.*;
+
+public class Parser {
+    public static Command parse(String message, Ui ui) throws LeluException {
+        if (message.equals("bye")) {
+            return new ByeCommand();
+        } else if (message.equals("list")) {
+            return new ListCommand();
+        }
+        switch (message.split(" ")[0]) {
+            case "mark":
+                return new MarkCommand();
+            case "unmark":
+                return new UnmarkCommand();
+            case "delete":
+                return new DeleteCommand();
+            case "todo":
+                return new AddToDoCommand();
+            case "deadline":
+                return new AddDeadlineCommand();
+            case "event":
+                return new AddEventCommand();
+            default:
+                throw new InvalidInputException(ui.showInstructions());
+        }
```
I like how you do the parse method, it is very clean and concise.

nigel27022001 commented on src/main/java/Tasks/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+
+    }
+
+    public static Deadline DeadlineOf(String input) throws InvalidInputException {
```
Any reason why the checking of deadline details is done here? Not sure I like that the checking of details is done here instead of the AddDeadlineCommand or Parser. On top of that, the constructor of Deadline is also public. Same for the Event and ToDo classes.

nigel27022001 reviewed (commented):
Good Job! I like how almost your entire code is really clean and efficient.  Just note of some minor issues such as having both a public constructor and a public method that constructs the same type of Object.

Lalelulilulela committed:
Add header comments to package Commands, Tasks, and Exceptions

Lalelulilulela committed:
Update package names

Lalelulilulela committed:
Update Package names

Lalelulilulela committed:
Update code to comply with coding standards

Lalelulilulela committed:
Add FindCommand.java

Lalelulilulela committed:
Merge branch 'branch-A-JavaDoc'

Lalelulilulela committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Exceptions/InvalidFormatException.java
#	src/main/java/exceptions/LeluException.java

Resolution: keep both

Lalelulilulela committed:
Remove unused file

Lalelulilulela committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/lelu.txt
#	src/main/java/Exceptions/InvalidFormatException.java
#	src/main/java/Lelu/Parser.java
#	src/main/java/Tasks/Task.java

Resolution: Keep Both

Lalelulilulela committed:
Fix merge conflict issues

Lalelulilulela committed:
Remove redundant methods in Deadline, Event and ToDo classes

Lalelulilulela committed:
Add checkstyle

chuahjiajie commented on src/main/java/UI/Ui.java:
```diff
@@ -21,7 +21,7 @@ public String showInstructions() {
 
     public void greet() {
-        String greet = "    Hi! I am your favourite friend, Lelu.Lelu :)\n    What can I do for you?\n";
+        String greet = "    Hi! I am your favourite friend, Lelu :)\n    What can I do for you?\n";
```
I think Lelu. Lelu is a nicer name.

chuahjiajie commented on data/lelu.txt:
```diff
@@ -1,3 +1 @@
-T | 1 | sleep        
-D | 0 | wake up  | tmr
-E | 1 | sleep  | 9pm  | 10am
```
Who sleeps at 9pm? Then wakes up at 10am (which is 11 hours of sleep??). R u ok?

chuahjiajie commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,53 @@
+import Tasks.Task;
```
Smart way of making UI, will use for inspiration thanks

chuahjiajie reviewed (commented):
The code is excellent. Lelu will be a great hit. ChatGPT more like GoneGPT 💀💀💀

Lalelulilulela committed:
Add GUI

NatLeong commented on src/main/java/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,40 @@
+ * the deadline task details and adds it to the task list, along with the specified
+ * due date and time.
+ */
+ public class AddDeadlineCommand extends Command {
```
Fantastic abstraction of Commands

NatLeong reviewed (commented):
Use of abstraction

NatLeong commented on src/main/java/exceptions/InvalidFormatException.java:
```diff
@@ -0,0 +1,37 @@
+     * @param task type of task with the wrong format.
+     * @throws InvalidFormatException Thrown with different error messages depending on the task type.
+     */
+    public static void callInvalidFormatException(ErrorType task) throws InvalidFormatException {
```
Good job on providing more specific errors for user guidance :)

NatLeong reviewed (commented):
Specific error for instruction format

NatLeong commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,39 @@
+
+public class Ui {
+
+    public String showInstructions() {
```
Clear user guidance and support. As a user, I would much appreciate this!!

NatLeong reviewed (commented):
Clear instructions

NatLeong commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,65 @@
+                String[] task = s.split(" \\| ");
+                Task t;
+                switch(task[0]) {
+                case "T":
```
I did not think of using the first letter as switch cases. Will take as inspiration!!

NatLeong reviewed (commented):
Switch cases for tasks

Lalelulilulela committed:
Update tests

Lalelulilulela committed:
Add assertions in the Commands Package

Assertions are enabled in build.gradle.

Assertions are added in various classes in the commands package to check if user input is handled appropriately.
This will help in verifying assumptions in the code.

Lalelulilulela committed:
Merge pull request #2 from Lalelulilulela/branch-A-Assertions

Add assertions in the Commands Package

Lalelulilulela committed:
Merge branch 'master' into branch-A-CodeQuality

Lalelulilulela committed:
Improve code quality

Lalelulilulela committed:
Merge pull request #3 from Lalelulilulela/branch-A-CodeQuality

Improve code quality

Lalelulilulela committed:
Add UpdateCommand extension

Lalelulilulela committed:
Add help extension

Lalelulilulela committed:
Merge branch 'branch-BCD-Extension'

Lalelulilulela committed:
Add Ui.png and fix bugs

Lalelulilulela committed:
Update README.md

Lalelulilulela committed:
Update README.md

Lalelulilulela committed:
Update README.md

Lalelulilulela committed:
Update README.md

Lalelulilulela committed:
Update Ui.png

Lalelulilulela committed:
Merge branch 'master' of github.com:Lalelulilulela/ip

Lalelulilulela committed:
Add more javadoc comments

Lalelulilulela committed:
Update Event.java

Lalelulilulela committed:
Update README.md
